hapi-fees/src/main/java/com/hedera/services/pricing/AssetsLoader.java

```diff
@@ -0,0 +1,129 @@
+package com.hedera.services.pricing;
+
+/*-
+ * ‌
+ * Hedera Services API Fees
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.SubType;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.EnumMap;
+import java.util.Map;
+
+/**
+ * Loads assets used to generate a fee schedule from JSON resources on the classpath.
+ *
+ * Please see the individual methods for details.
+ */
+class AssetsLoader {
+	private static final String CAPACITIES_RESOURCE = "capacities.json";
+	private static final String CONSTANT_WEIGHTS_RESOURCE = "constant-weights.json";
+	private static final String CANONICAL_PRICES_RESOURCE = "canonical-prices.json";
+
+	/**
+	 * Loads a map that, for each supported operation, gives the fraction of that
+	 * operation's total price that should come from its constant term in the fee
+	 * schedule.
+	 *
+	 * This fraction is the "weight" of the constant term; and is currently set at
+	 * 0.9 for operations that create new entities, and 0.2 for other operations.
+	 *
+	 * @return the "weight" of the constant term for each operation
+	 * @throws IOException if the backing JSON resource cannot be loaded
+	 */
+	Map<HederaFunctionality, BigDecimal> loadConstWeights() throws IOException {
+		try (var fin = AssetsLoader.class.getClassLoader().getResourceAsStream(CONSTANT_WEIGHTS_RESOURCE)) {
+			final var om = new ObjectMapper();
+			final var constWeights = om.readValue(fin, Map.class);
+
+			final Map<HederaFunctionality, BigDecimal> typedConstWeights = new EnumMap<>(HederaFunctionality.class);
+			constWeights.forEach((funcName, weight) -> {
+				final var function = HederaFunctionality.valueOf((String) funcName);
+				final var bdWeight = (weight instanceof Double)
+						? BigDecimal.valueOf((Double) weight)
+						: BigDecimal.valueOf((Integer) weight);
```

Comment:
nit: can we require the data source always to be provided as decimal format, so no need to consider the `Integer` case?

Reply:
Agreed, no need for this check; removed.